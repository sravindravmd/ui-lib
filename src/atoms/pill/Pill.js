import React from 'react'
import PropTypes from 'prop-types'

import { StyledPill, pillTypes } from './Pill.styles'

export const Pill = ({ variant, text, ...props }) => {
  const pillType = pillTypes[variant]
    ? pillTypes[variant]
    : pillTypes['success']

  return (
    <StyledPill {...props} {...pillType}>
      {text}
    </StyledPill>
  )
}

Pill.defaultProps = {
  variant: 'success',
}

Pill.propTypes = {
  text: PropTypes.string.isRequired,
  variant: PropTypes.oneOf(['success', 'warning', 'alert']),
}
