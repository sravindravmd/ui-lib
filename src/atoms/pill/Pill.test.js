import React from 'react'
import { render, cleanup } from '@testing-library/react'
import { Pill } from './Pill'

afterEach(cleanup)

describe('<Pill/>', () => {
  it('renders Pill component with variant:success correctly', () => {
    const { container } = render(<Pill variant='success' text='Pill Label' />)
    expect(container).toMatchSnapshot()
  })

  it('renders Pill component with variant:warning correctly', () => {
    const { container } = render(<Pill variant='warning' text='Pill Label' />)
    expect(container).toMatchSnapshot()
  })

  it('renders Pill component with variant:alert correctly', () => {
    const { container } = render(<Pill variant='alert' text='Pill Label' />)
    expect(container).toMatchSnapshot()
  })
})
