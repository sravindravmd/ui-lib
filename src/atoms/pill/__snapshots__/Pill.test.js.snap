// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Pill/> renders Pill component with variant:alert correctly 1`] = `
<div>
  <span
    class="sc-bdfBwQ fFEcdn"
  >
    Pill Label
  </span>
</div>
`;

exports[`<Pill/> renders Pill component with variant:success correctly 1`] = `
<div>
  <span
    class="sc-bdfBwQ hjkiSL"
  >
    Pill Label
  </span>
</div>
`;

exports[`<Pill/> renders Pill component with variant:warning correctly 1`] = `
<div>
  <span
    class="sc-bdfBwQ emjRfH"
  >
    Pill Label
  </span>
</div>
`;
